File: java/src/main/java/challenges/aoc2024/day4/Application.java
package challenges.aoc2024.day4;

import challenges.aoc2024.day4.service.DayFourSolver;
import challenges.aoc2024.day4.service.WordsearchService;
import service.TaskSolver;

public class Application {
    public static void main(String[] args) {
        final TaskSolver solver = new TaskSolver(new DayFourSolver(new WordsearchService()));
        solver.solveTasks();
    }
}

File: scala/src/main/scala/challenges/aoc2024/day4/Application.scala
package challenges.aoc2024.day4

import _root_.service.TaskSolver
import challenges.aoc2024.day4.service.{DayFourSolver, WordsearchService}

object Application {
  def main(args: Array[String]): Unit = {
    val solver: TaskSolver = TaskSolver(DayFourSolver(WordsearchService()))
    solver.solveTasks()
  }
}
